addons/event_booth/models/event_booth_slot.py

```diff
@@ -0,0 +1,159 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import pytz
+
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
+from odoo.tools import format_datetime
+
+
+class EventBoothSlot(models.Model):
+    """Even Booth Slot
+
+    Event Booths are split into one or several Slots on which the event manager define a date range.
+    Users will be able to register to one ore more slots they want to rent during the event.
+    """
+    _name = 'event.booth.slot'
+    _description = 'Event Booth Slot'
+    _order = 'booking_from asc'
+
+    # Event / Description
+    event_booth_id = fields.Many2one('event.booth', string='Booth', readonly=True, required=True)
+    event_id = fields.Many2one(string='Event', related='event_booth_id.event_id', store=True)
+    event_date_begin = fields.Datetime(
+        string='Event Start Date', related='event_id.date_begin')
+    event_date_end = fields.Datetime(
+        string='Event End Date', related='event_id.date_end')
+    booking_from = fields.Datetime(
+        string='Start Date', required=True)
+    booking_to = fields.Datetime(
+        string='End Date', required=True)
+    state = fields.Selection([
+        ('available', 'Available'),
+        ('reserved', 'Reserved'),
+        ('unavailable', 'Unavailable'),
+    ], string='Status', default='available', required=True)
+    # Contact Information
+    partner_id = fields.Many2one('res.partner', string='Rented By')
+    topic = fields.Char(string='Topic')
+    name = fields.Char(string='Name')
+    email = fields.Char(string='Email')
+    phone = fields.Char(string='Phone')
+    mobile = fields.Char(string='Mobile')
```

Comment:
why not computed based on partner but still readonly=False, store=True in order to override if needed ?
well.. maybe not necessary indeed.. mmmh :thinking: